Makefile

```diff
@@ -1,768 +0,0 @@
-
-%_gpu.o: %.cu
-	@mkdir -p $(@D)
-	$(NVCC) $(NVCCFLAGS) $(CUDA_ARCH) -Xcompiler "$(CFLAGS) -Isrc/operator" --generate-dependencies -MT $*_gpu.o $< >$*_gpu.d
-	$(NVCC) -c -o $@ $(NVCCFLAGS) $(CUDA_ARCH) -Xcompiler "$(CFLAGS) -Isrc/operator" $<
-
-%.o: %.cc $(CORE_INC)
-	@mkdir -p $(@D)
-	$(CXX) -std=c++17 -c $(CFLAGS) -MMD -Isrc/operator -c $< -o $@
-
-# Set install path for libmxnet.so on Mac OS
-ifeq ($(UNAME_S), Darwin)
-        LDFLAGS += -Wl,-install_name,@rpath/libmxnet.so
-endif
-
-# NOTE: to statically link libmxnet.a we need the option
-# --Wl,--whole-archive -lmxnet --Wl,--no-whole-archive
-lib/libmxnet.a: $(ALLX_DEP)
-	@mkdir -p $(@D)
-	ar crv $@ $(filter %.o, $?)
-
-lib/libmxnet.so: $(ALLX_DEP)
-	@mkdir -p $(@D)
-	$(CXX) $(CFLAGS) -shared -o $@ $(filter-out %libnnvm.a, $(filter %.o %.a, $^)) $(LDFLAGS) \
-	-Wl,${WHOLE_ARCH} $(filter %libnnvm.a, $^) -Wl,${NO_WHOLE_ARCH}
-
-$(PS_PATH)/build/libps.a: PSLITE
-
-PSLITE:
-	$(MAKE) CXX="$(CXX)" DEPS_PATH="$(DEPS_PATH)" -C $(PS_PATH) ps
-
-$(DMLC_CORE)/libdmlc.a: DMLCCORE
-
-DMLCCORE:
-	+ cd $(DMLC_CORE); $(MAKE) libdmlc.a USE_SSE=$(USE_SSE) config=$(ROOTDIR)/$(config); cd $(ROOTDIR)
-
-lib/libtvm_runtime.so:
-	echo "Compile TVM"
-	@mkdir -p $(@D)
-	[ -e $(LLVM_PATH)/bin/llvm-config ] || sh $(ROOTDIR)/contrib/tvmop/prepare_tvm.sh; \
-	cd $(TVM_PATH)/build; \
-	cmake -DUSE_LLVM="$(LLVM_PATH)/bin/llvm-config" \
-		  -DUSE_SORT=OFF -DUSE_CUDA=$(TVM_USE_CUDA) -DUSE_CUDNN=OFF -DUSE_OPENMP=ON ..; \
-	$(MAKE) VERBOSE=1; \
-	mkdir -p $(ROOTDIR)/lib; \
-	cp $(TVM_PATH)/build/libtvm_runtime.so $(ROOTDIR)/lib/libtvm_runtime.so; \
-	ls $(ROOTDIR)/lib; \
-	cd $(ROOTDIR)
-
-TVM_OP_COMPILE_OPTIONS = -o $(ROOTDIR)/lib --config $(ROOTDIR)/lib/tvmop.conf
-ifneq ($(CUDA_ARCH),)
-	TVM_OP_COMPILE_OPTIONS += --cuda-arch "$(CUDA_ARCH)"
-endif
-lib/libtvmop.so: lib/libtvm_runtime.so $(wildcard contrib/tvmop/*/*.py contrib/tvmop/*.py)
-	echo "Compile TVM operators"
-	@mkdir -p $(@D)
-	PYTHONPATH=$(TVM_PATH)/python:$(TVM_PATH)/topi/python:$(ROOTDIR)/contrib \
-		LD_LIBRARY_PATH=$(ROOTDIR)/lib \
-	    python3 $(ROOTDIR)/contrib/tvmop/compile.py $(TVM_OP_COMPILE_OPTIONS)
-
-NNVM_INC = $(wildcard $(NNVM_PATH)/include/*/*.h)
-NNVM_SRC = $(wildcard $(NNVM_PATH)/src/*/*/*.cc $(NNVM_PATH)/src/*/*.cc $(NNVM_PATH)/src/*.cc)
-$(NNVM_PATH)/lib/libnnvm.a: $(NNVM_INC) $(NNVM_SRC)
-	+ cd $(NNVM_PATH); $(MAKE) lib/libnnvm.a DMLC_CORE_PATH=$(DMLC_CORE); cd $(ROOTDIR)
-
-bin/im2rec: tools/im2rec.cc $(ALLX_DEP)
-
-$(BIN) :
-	@mkdir -p $(@D)
-	$(CXX) $(CFLAGS) -std=c++17  -o $@ $(filter %.cpp %.o %.c %.a %.cc, $^) $(LDFLAGS)
-
-include mkldnn.mk
-include tests/cpp/unittest.mk
-
-extra-packages: $(EXTRA_PACKAGES)
-
-test: $(TEST)
-
-lint: cpplint pylint
-
-cpplint:
-	3rdparty/dmlc-core/scripts/lint.py mxnet cpp include src plugin tests \
-	--exclude_path src/operator/contrib/ctc_include include/mkldnn
-
-pylint:
-	python3 -m pylint --rcfile=$(ROOTDIR)/ci/other/pylintrc --ignore-patterns=".*\.so$$,.*\.dll$$,.*\.dylib$$" python/mxnet
```

Comment:
should we add these somewhere in the cmake?

Reply:
better yet if we could have them as a git hook before check-in

Reply:
Developers may run `python ci/build.py --platform ubuntu_cpu /work/runtime_functions.sh sanity_check` to reproduce the CI Sanity check. We can provide git hooks in a separate effort